wcfsetup/install/files/ts/WoltLabSuite/Core/Prism/Helper.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Provide helper functions for prism processing.
+ *
+ * @author	Tim Duesterhus
+ * @copyright	2001-2021 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module	WoltLabSuite/Core/Prism/Helper
+ */
+
+export function* splitIntoLines(container: Node): Generator<Element, void> {
+  const it = document.createNodeIterator(container, NodeFilter.SHOW_TEXT, {
+    acceptNode() {
+      return NodeFilter.FILTER_ACCEPT;
+    },
+  });
+
+  let line = document.createElement("span");
+  for (let node = it.nextNode(); node; node = it.nextNode()) {
```

Comment:
Isn't this just a `while (let node = it.nextNode())` in disguise? I can see why `for (;;)` is more concise than `while (true)`, but in this case the `for()` implies a known sequence, which it isn't. Maybe I'm just a bit nitpicky here.